app/assets/javascripts/school_administrator_dashboard/discipline_dashboard/SchoolDisciplineDashboard.js

```diff
@@ -186,22 +235,10 @@ export default class SchoolDisciplineDashboard extends React.Component {
   }
 
   renderStudentDisciplineTable(allIncidents, groupedIncidents) {
-    const students = this.groupStudents();
-    const studentDisciplineIncidentCounts = this.state.selectedCategory ? //if the user is looking at a subgroup of incidents
-                                            this.studentDisciplineIncidentCounts(groupedIncidents[this.state.selectedCategory]) :
-                                            this.studentDisciplineIncidentCounts(allIncidents);
-    let rows =[];
-    students.forEach((student) => {
-      rows.push({
-        id: student.id,
-        first_name: student.first_name,
-        last_name: student.last_name,
-        latest_note: student.latest_note,
-        events: studentDisciplineIncidentCounts[student.id] || 0,
-        grade: student.grade
-      });
-    });
-
+    const selectedIncidents = this.state.selectedCategory ? //if the user is looking at a subgroup of incidents
```

Comment:
I think this branch and grouping is the kind of thing we want to remove, and isolate in methods like `filteredStudents` or `filteredDisciplineIncidents`.  The way it is now, if we want to add a filter for grade next week, we'd have to rework this and make the index more complex to support that.  If we can isolate the filtering, then updating filters just requires updating the filter function.

Could this method work with calling `filteredStudents` or `filteredIncidents`?